internal/databases/mysql/binlog_push_handler.go

```diff
@@ -53,47 +53,67 @@ func HandleBinlogPush(uploader internal.UploaderProvider, untilBinlog string, ch
 		tracelog.InfoLogger.Printf("fetched binlog archived GTID SET: %s\n", cache.GTIDArchived)
 	}
 
-	var filters []statefulBinlogFilter
-	filters = append(filters, &untilBinlogFilter{Until: untilBinlog}, &archivedBinlogFilter{})
+	var filter gtidFilter
 	if checkGTIDs {
 		flavor, err := getMySQLFlavor(db)
 		if flavor == "" || err != nil {
 			flavor = mysql.MySQLFlavor
 		}
 		if flavor == mysql.MySQLFlavor {
-			filters = append(filters, &gtidFilter{BinlogsFolder: binlogsFolder, Flavor: flavor})
+			gtid, _ := mysql.ParseMysqlGTIDSet(cache.GTIDArchived)
+			gtidArchived, _ := gtid.(*mysql.MysqlGTIDSet)
```

Comment:
It's not safe in conjunction with flavor defaults on L60 ^^.
Let's return error if we are not sure with flavor.